inc/user.class.php

```diff
@@ -4499,23 +4499,23 @@ static function getUniquePersonalToken() {
    /**
     * Get token of a user. If not exists generate it.
     *
-    * @param $ID user ID
+    * @param $ID no longer used
     * @param string $field the field storing the token
     *
     * @return string token
    **/
-   static function getToken($ID, $field = 'personal_token') {
+   function getToken($ID = null, $field = 'personal_token') {
```

Comment:
> I think we have too keep original method and get it deprecated.
We already have getPersonalToken which is deprecated, and getToken allows to get the API or the personal token. Is it a good idea to deprecate getToken and add again a new one ?

getToken() was added recently (GLPI 9.2), right ?


> Also, since the $ID will be used less than $field; $field should be first.

By choosing to keep the getToken() method and planning a deprecation phase, it is better to keep prototype unchanged.

Conclusion: create a new method is the easy way, but I don't have inspiration to find a good name yet :)

Reply:
I may create ```getUserToken($field = 'personal_token')``` OK with this name ?